pkg/controller/controlplane/valuesprovider.go

```diff
@@ -272,22 +273,27 @@ func (vp *valuesProvider) GetStorageClassesChartValues(
 		}
 	}
 	values := make(map[string]interface{})
-	if providerConfig.StorageClasses != nil && len(providerConfig.StorageClasses) != 0 {
-		allSc := make([]map[string]interface{}, len(providerConfig.StorageClasses))
-		for i, sc := range providerConfig.StorageClasses {
-			var storageClassValues = map[string]interface{}{
-				"name": sc.Name,
-				"type": sc.Type,
-			}
-
-			if sc.Default != nil && *sc.Default {
-				storageClassValues["default"] = true
-			}
-
-			allSc[i] = storageClassValues
+	allScs := make([]map[string]interface{}, len(providerConfig.StorageClasses.AdditionalStorageClasses)+1)
+	defaultSc := make(map[string]interface{})
```

Comment:
Please just append directly the default storage class to the output map if it is defined. No need to have `defaultSc` here.

Reply:
Done